.github/workflows/build_and_test.yml

```
@@ -683,6 +683,9 @@ jobs:
               \"description\": \"[$REMOTE_REPO] Remote CI job is about to start\"
             }"
 
+          # Pass current concurrency group to the e2e-tests CI job to make sure e2e-tests job won't have unneeded parallel runs
+          CONCURRENCY_GROUP="${{ github.workflow }}-${{ github.ref_name }}-${{ github.ref_name == 'main' && github.sha || 'anysha' }}"
```

By my reasoning this will allow:
- multiple main runs
- 1 PR run

So looking good.

Should we add a prefix to ensure no collisions with the workflow names between repos? Those would be hairy to debug.
```suggestion
          CONCURRENCY_GROUP="neon-${{ github.workflow }}-${{ github.ref_name }}-${{ github.ref_name == 'main' && github.sha || 'anysha' }}"
```

---

Prefixed with `${{ github.repository }}` and moved to a global `env` since it's going to be used in 2 workflows

---

Makes a lot of sense, this is looking good.